catalog/core/catalog-core-api/src/main/java/ddf/catalog/content/operation/ContentMetadataExtractor.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package ddf.catalog.content.operation;
+
+import java.util.Set;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+
+/**
+ * Parses content, extracting metadata and adding it to the provided card.
+ */
+public interface ContentMetadataExtractor {
```

Comment:
Might be a nitpick, but I find this interface name a little bit confusing. It's not actually parsing content, but rather text that has been extracted from content. Something like `TextMetadataExtractor` would make more sense to me, since that's what the interface really is. It takes in some unstructured text, extracts metadata, and puts it on the metacard.